--- hostname/Makefile.orig	2017-12-19 20:04:49.422378235 +0000
+++ hostname/Makefile	2017-12-19 20:05:15.183377565 +0000
@@ -10,21 +10,17 @@
 
 hostname: $(OBJS)
 	$(CC) $(CFLAGS) -o $@ $(OBJS) $(LDFLAGS)
-	ln -fs hostname dnsdomainname
-	ln -fs hostname domainname
-	ln -fs hostname ypdomainname
-	ln -fs hostname nisdomainname
 
 install: hostname
 	install -d ${BASEDIR}$(MANDIR)/man1
-	install -o root -g root -m 0644 hostname.1 ${BASEDIR}$(MANDIR)/man1
+	install -m 0644 hostname.1 ${BASEDIR}$(MANDIR)/man1
 	ln -fs hostname.1 ${BASEDIR}$(MANDIR)/man1/dnsdomainname.1
 	ln -fs hostname.1 ${BASEDIR}$(MANDIR)/man1/domainname.1
 	ln -fs hostname.1 ${BASEDIR}$(MANDIR)/man1/ypdomainname.1
 	ln -fs hostname.1 ${BASEDIR}$(MANDIR)/man1/nisdomainname.1
 
 	install -d ${BASEDIR}$(BINDIR)
-	install -o root -g root -m 0755 hostname ${BASEDIR}$(BINDIR)
+	install -m 0755 hostname ${BASEDIR}$(BINDIR)
 	ln -fs hostname ${BASEDIR}$(BINDIR)/dnsdomainname
 	ln -fs hostname ${BASEDIR}$(BINDIR)/domainname
 	ln -fs hostname ${BASEDIR}$(BINDIR)/nisdomainname
