--- riscv-binutils-gdb-riscv-binutils-2.29/gdb/proc-service.c~	2017-12-04 22:15:15.000000000 +0000
+++ riscv-binutils-gdb-riscv-binutils-2.29/gdb/proc-service.c	2017-12-19 18:29:07.171826816 +0000
@@ -157,7 +157,7 @@
     = get_thread_arch_regcache (ptid, target_gdbarch ());
 
   target_fetch_registers (regcache, -1);
-  fill_gregset (regcache, (gdb_gregset_t *) gregset, -1);
+  //fill_gregset (regcache, (gdb_gregset_t *) gregset, -1);
 
   return PS_OK;
 }
@@ -172,7 +172,7 @@
   struct regcache *regcache
     = get_thread_arch_regcache (ptid, target_gdbarch ());
 
-  supply_gregset (regcache, (const gdb_gregset_t *) gregset);
+  //supply_gregset (regcache, (const gdb_gregset_t *) gregset);
   target_store_registers (regcache, -1);
 
   return PS_OK;
@@ -190,7 +190,7 @@
     = get_thread_arch_regcache (ptid, target_gdbarch ());
 
   target_fetch_registers (regcache, -1);
-  fill_fpregset (regcache, (gdb_fpregset_t *) fpregset, -1);
+  //fill_fpregset (regcache, (gdb_fpregset_t *) fpregset, -1);
 
   return PS_OK;
 }
@@ -206,7 +206,7 @@
   struct regcache *regcache
     = get_thread_arch_regcache (ptid, target_gdbarch ());
 
-  supply_fpregset (regcache, (const gdb_fpregset_t *) fpregset);
+  //supply_fpregset (regcache, (const gdb_fpregset_t *) fpregset);
   target_store_registers (regcache, -1);
 
   return PS_OK;
