--- riscv-gnu-toolchain-728afcddcb0526a0f6560c4032da82805f054d58/Makefile.in.old	2016-10-27 09:06:32.208278427 +0100
+++ riscv-gnu-toolchain-728afcddcb0526a0f6560c4032da82805f054d58/Makefile.in	2016-10-27 09:06:43.078270374 +0100
@@ -209,7 +209,7 @@
 		--with-sysroot=$(SYSROOT) \
 		--enable-shared \
 		--enable-tls \
-		--enable-languages=c,c++,fortran \
+		--enable-languages=c,c++,fortran,ada \
 		--disable-libmudflap \
 		--disable-libssp \
 		--disable-libquadmath \
