From 39c1972db2255b6e38b7ae32bf343966b4422796 Mon Sep 17 00:00:00 2001
From: "Richard W.M. Jones" <rjones@redhat.com>
Date: Wed, 27 Dec 2017 20:00:41 +0000
Subject: [PATCH 2/2] target-riscv: Print PC when handling EBREAK.

---
 target/riscv/helper.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/target/riscv/helper.c b/target/riscv/helper.c
index 34da7ddbf2..525b08e46a 100644
--- a/target/riscv/helper.c
+++ b/target/riscv/helper.c
@@ -392,7 +392,8 @@ void riscv_cpu_do_interrupt(CPUState *cs)
     #endif
 
     if (cs->exception_index == RISCV_EXCP_BREAKPOINT) {
-        fprintf(stderr, "debug mode not implemented\n");
+        fprintf(stderr, "debug mode not implemented, "
+                "pc = 0x" TARGET_FMT_lx "\n", env->pc);
     }
 
     /* skip dcsr cause check */
-- 
2.15.1

